### ISSUE TITLE
Figure out some way to attribute end crystal pvp deaths to the correct players.

### ISSUE THREAD

**Issue by Jaxkr** [Jaxkr/mammoth#42]
This is a tricky issue even when you're not dealing with cross-server combat.

https://github.com/plan-player-analytics/Plan/issues/1571
https://www.spigotmc.org/threads/ender-crystal-get-killer.474526/

We need this in order to drop blocks for the person who detonated the end crystal that dealt the killing blow.

**Comment by walksanatora** [comment]
i thought end crystals didn't track who placed/exploded them (Source: entirety of mumbo jumbo ruing season 8 of hermit craft where he refued to have a single kill)

**Comment by Jaxkr** [comment]
> i thought end crystals didn't track who placed/exploded them (Source: entirety of mumbo jumbo ruing season 8 of hermit craft where he refued to have a single kill)

Yep, you're exactly correct. We can figure out when they're created by listening for a PlayerInteractEvent then searching for a new nearby end crystal after a few ticks. Attributing them to a player, especially cross-server, is a bit of a mystery.

**Comment by fatpowaranga** [comment]
Just a thought as I’m still familiarising myself with Mammoth, but would it not be enough to attribute to the server where the explosion event originated from? A play through of events:

1. End Crystal is placed in Server A
2. Player A detonates End Crystal from Server B
3. Both Player B and Player C were killed by explosion on Server A and Server B respectively.
4. Loot from both player B and Player C drops on Server B.

Apologies if I’m misunderstanding

**Comment by AuroraLS3** [comment]
You can find who killed the end crystal from the last entity that caused damage to it - although since you have multiple servers I'd imagine you need some netcode to get that information between servers.

If you use some sort of payload for exploding the crystal on server B you can lookup the killer of crystal on server A and include the information before sending the payload
